system/ee/legacy/libraries/Template.php

```diff
@@ -232,15 +232,40 @@ public function fetch_and_parse($template_group = '', $template = '', $is_embed
         //  - Modify template after tag parsing
         //
         if (ee()->extensions->active_hook('template_post_parse') === true) {
-            $this->final_template = ee()->extensions->call(
-                'template_post_parse',
-                $this->final_template,
-                ($is_embed || $is_layout), // $is_partial
-                $site_id
-            );
+            // Populate the $currentTemplateInfo array
+            $currentTemplateInfo = array();
+            if ($this->templates_loaded) // don't do this if we don't have any template info!
+            {
+                $currentTemplateInfo = array(
+                    'template_name' => $template,
+                    'template_group' => $template_group,
+                );
+            }
+            // Add a conditional to adjust what is returned by function depending on what kind of template we are dealing with
+            if ($this->final_template)
+            {
+                $this->final_template = ee()->extensions->call(
+                    'template_post_parse',
+                    $this->final_template,
+                    ($is_embed || $is_layout), // $is_partial
+                    $site_id,
+                    $currentTemplateInfo
+                );
+            }
+            else
+            {
+                $this->template = ee()->extensions->call(
+                    'template_post_parse',
+                    $this->template,
+                    ($is_embed || $is_layout), // $is_partial
+                    $site_id,
+                    $currentTemplateInfo
+                );
+            }
         }
         //
         // -------------------------------------------
+    return;
```

Comment:
What is the reason this `return` has been added any why it does not return any value?

Reply:
Simply to provide a point to add a debugging breakpoint.  Can be removed.